### ISSUE TITLE
Required attributes serialization\deserialization

### ISSUE THREAD

**Issue by bidolah** [RHEAGROUP/reqifsharp#3]
Hi!

I looked to the reqif schema and find that MAX, MIN attributes is required in DATATYPE-DEFINITION-INTEGER (and other...)

![image](https://user-images.githubusercontent.com/9832538/40892865-0668d9de-67a5-11e8-9df0-fc245bd56c79.png)

Should we throw exception for serialization\deserialization if field data is not filled out or has wrong value?

**Comment by samatrhea** [comment]
The ReqIFSharp library can be used with and without XML Schema validation. ReqIFSharp currently targets the .NET Framework 4.5 and netstandard 1.6. When using netstandard 1.6 it does not support XML Schema validation.  An existing ReqIF document can be deserialized to an object graph using the ReqIFDeserializer class that implements the IReqIFDeserializer interface. The IReqIFDeserializer interface exposes one method, the Deserialize method. 

``` 
/// <summary>
        /// Deserializes a <see cref="ReqIF"/> XML document.
        /// </summary>
        /// <param name="xmlFilePath">
        /// The Path of the <see cref="ReqIF"/> file to deserialize
        /// </param>
        /// <param name="validate">
        /// a value indicating whether the XML document needs to be validated or not
        /// </param>
        /// <param name="validationEventHandler">
        /// The <see cref="ValidationEventHandler"/> that processes the result of the <see cref="ReqIF"/> validation.
        /// </param>
        /// <returns>
        /// A fully dereferenced <see cref="ReqIF"/> object graph
        /// </returns>
ReqIF Deserialize(string xmlFilePath, bool validate = false, ValidationEventHandler validationEventHandler = null);
``` 
as you can see it can handle with and without schema validation. I propose we keep the logic in the schema validator and not implement it explicitly in the code.

**Comment by samatrhea** [comment]
You make a valid point, give me some time to think about it, will get back to you soon. @naron-rhea feel free to chime in once you come online.

**Comment by bidolah** [comment]
I recommend not to throw out the exceptions. So, I plan to do PR. Do you agree @naron-rhea  @samatrhea  ?

**Comment by naron-rhea** [comment]
@bidolah Yes I think it would be better not to throw the exception when parsing and leave that responsability to the validator.

**Comment by samatrhea** [comment]
there is a difference between parsing/deserialization (reading in a reqif document) and serialization/writing a document.

@bidolah: in your pull request most proposed updates are related to writing and not to reading.
